From: Ben Hutchings <ben@decadent.org.uk>
Date: Wed, 24 Sep 2014 03:02:28 +0100
Subject: sp5100_tco: Reject SB8x0 chips
Bug-Debian: https://bugs.debian.org/726150
Forwarded: not-needed

The SMBus functions of SB8x0 chips have the same PCI ID as on the
SP5100, but the TCO timer has a different register set!  They can be
distinguished by PCI revision.

Upstream commit 740fbddf5c3f ('watchdog: sp5100_tco: Add SB8x0 chipset
support') adds support for the newer chips, but we're playing safe
here by rejecting them completely.

--- a/drivers/watchdog/sp5100_tco.c
+++ b/drivers/watchdog/sp5100_tco.c
@@ -287,6 +287,10 @@ static unsigned char __devinit sp5100_tc
 	if (!sp5100_tco_pci)
 		return 0;
 
+	/* Higher revisions are SB8x0 which have a different register set */
+	if (sp5100_tco_pci->revision >= 0x40)
+		return 0;
+
 	/* Request the IO ports used by this driver */
 	pm_iobase = SP5100_IO_PM_INDEX_REG;
 	if (!request_region(pm_iobase, SP5100_PM_IOPORTS_SIZE, "SP5100 TCO")) {
